app/upgrades/v0.0.12.go

```diff
@@ -0,0 +1,130 @@
+			// GetParams() to avoid potential protobuf issues and all networks have the same value (no need to read).
+			// Validate with: `poktrolld q supplier params --node=https://testnet-validated-validator-rpc.poktroll.com/`
+			supplierParams := suppliertypes.Params{
+				MinStake: &cosmosTypes.Coin{
+					Denom:  "upokt",
+					Amount: math.NewInt(1000000),
+				},
+				StakingFee: &cosmosTypes.Coin{
+					Denom: "upokt",
+					// TODO_IN_THIS_PR: 100upokt a good value?
+					Amount: math.NewInt(100),
+				},
+			}
+
+			// ALL parameters must be present when setting params.
+			err = keepers.SupplierKeeper.SetParams(ctx, supplierParams)
+			if err != nil {
+				logger.Error("Failed to set supplier params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated supplier params", "new_params", supplierParams)
+
+			// Add service module `target_num_relays` parameter per `config.yml`.
+			// We don't use `GetParams()` to avoid potential protobuf issues and all networks have the same value (no need to read).
+			serviceParams := servicetypes.Params{
+				AddServiceFee: &cosmosTypes.Coin{
+					Denom:  "upokt",
+					Amount: math.NewInt(1000000000),
+				},
+				TargetNumRelays: 100,
+			}
+			err = keepers.ServiceKeeper.SetParams(ctx, serviceParams)
+			if err != nil {
+				logger.Error("Failed to set service params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated service params", "new_params", serviceParams)
+
+			// Add tokenomics module `global_inflation_per_claim` parameter per `config.yml`.
+			// We use GetParams() as `DaoRewardAddress` is different between networks and we don't want to hardcode it.
+			tokenomicsParams := keepers.TokenomicsKeeper.GetParams(ctx)
+			tokenomicsParams.GlobalInflationPerClaim = 0.1
+			err = keepers.TokenomicsKeeper.SetParams(ctx, tokenomicsParams)
+			if err != nil {
+				logger.Error("Failed to set tokenomics params", "error", err)
+				return err
+			}
+			return
+		}
+
+		// Returns the upgrade handler for v0.0.12
+		return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+			logger := cosmosTypes.UnwrapSDKContext(ctx).Logger()
+			logger.Info("Starting v0.0.12 upgrade handler")
+
+			err := applyNewParameters(ctx)
+			if err != nil {
+				logger.Error("Failed to apply new parameters", "error", err)
+				return vm, err
+			}
+
+			// Since we changed the type of RevSharePercent from float32 to uint64, we need to update all on-chain data.
+			// The easiest way to do this is to iterate over all suppliers and services and set the revshare to 100 by default.
+			suppliers := keepers.SupplierKeeper.GetAllSuppliers(ctx)
+			logger.Info("Updating all suppliers to have a 100% revshare to the supplier", "num_suppliers", len(suppliers))
+			for _, supplier := range suppliers {
+				for _, service := range supplier.Services {
+					// Force all services to have a 100% revshare to the supplier.
```

Comment:
Can you change this to read the prior state and use it for the new state/

I realize it's not a big deal here, but I want us to take this as an opportunity to do it right, and set a right example/reference in the future.

Reply:
Can you please elaborate? `GetAllSuppliers` gives us the current state from what I understand.

I do want to add a warning if there's a supplier with more than 1 revshare, which would signal us if we broke somebody's setup.